apps/fz_http/lib/fz_http_web/live/user_live/show_live.ex

```diff
@@ -63,4 +63,56 @@ defmodule FzHttpWeb.UserLive.Show do
       end
     end
   end
+
+  @impl Phoenix.LiveView
+  def handle_event(action, %{"user_id" => user_id}, socket) when action in ~w(promote demote) do
+    if user_id == "#{socket.assigns.current_user.id}" do
+      {:noreply,
+       socket
+       |> put_flash(:error, "Changing your own role is not supported.")}
+    else
+      user = Users.get_user!(user_id)
+
+      role =
+        case action do
+          "promote" -> :admin
+          "demote" -> :unprivileged
+        end
+
+      case Users.update_user(user, %{role: role}) do
+        {:ok, user} ->
+          # Force reconnect with new role
+          FzHttpWeb.Endpoint.broadcast("users_socket:#{user.id}", "disconnect", %{})
+
+          {:noreply,
+           socket
+           |> assign(:user, user)
+           |> put_flash(:info, "User updated successfully.")}
+
+        {:error, changeset} ->
+          {:noreply,
+           socket
+           |> put_flash(
+             :error,
+             "Error updating user: #{ErrorHelpers.aggregated_errors(changeset)}"
+           )}
+      end
+    end
+  end
+
+  def mote(%{role: :admin}) do
+    "demote"
+  end
+
+  def mote(%{role: _}) do
+    "promote"
+  end
+
+  def mote_message(%{role: :admin}) do
+    "This will remove admin permissions from the user."
+  end
+
+  def mote_message(%{role: _}) do
+    "This will give admin permissions to the user."
+  end
```

Comment:
Need help with a better name :)

Also, are they supposed to be here? Or is there a better place in live view structure?

Reply:
This looks good for now. Should these be private functions, or were you planning to unit test them?
Also, should we make these messages module attributes?

Reply:
Refactored this part. Somehow I was under the impression that private functions are not accessible in the template.